### PR TITLE
[sc-10868] refactor prompt lab to rag lab

### PR DESCRIPTION
**Pull request by mpellerin42**
- Prompt lab component stay mostly unchanged (from a user perspective) but is now displayed as a tab of the new RAG lab page
- the RAG lab tab allows to compare the results for the search configurations saved by the user
- Move the results generation and management logic in a service
- Create a lab-layout component for the common parts of prompt and RAG labs.

### CODE REVIEW

File: libs/common/src/assets/i18n/en.json [hunk 1]
```diff
@@ -780,34 +780,41 @@
...
+	"rag-lab.common.results.download-csv": "Download CSV",
+	"rag-lab.common.results.error": "An error occurred while getting the answer, please try again later.",
+	"rag-lab.common.results.prompt": "Prompt: ",
+	"rag-lab.common.results.question": "Question: {{query}}",
+	"rag-lab.common.results.title": "Results",
+	"rag-lab.page-description": "Test different search configurations and RAG strategies, choose which settings you want to apply to your Knowledge Box.",
+	"rag-lab.page-title": "RAG lab",
+	"rag-lab.prompt-lab.configuration.models.title": "Select all the generative models you want to compare",
+	"rag-lab.prompt-lab.configuration.prompt.example": "Example",
+	"rag-lab.prompt-lab.configuration.prompt.examples.first-example": "Taking into account this context:\n{context}\nGive a detailed answer to this question:\n{question} . \nIf you do not find an answer in the context, say that you don't have enough data.",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/en.json:798]:
```suggestion
	"rag-lab.prompt-lab.configuration.prompt.examples.first-example": "Taking into account this context:\n{context}\nGive a detailed answer to this question:\n{question} . \nIf you do not find an answer in the context, say that you don’t have enough data.",
```

File: libs/common/src/assets/i18n/en.json [hunk 2]
```diff
@@ -780,34 +780,41 @@
...
+	"rag-lab.prompt-lab.configuration.system-prompt.label": "System prompt",
+	"rag-lab.prompt-lab.configuration.system-prompt.placeholder": "Define the personality traits of the assistant",
+	"rag-lab.prompt-lab.configuration.title": "Configuration",
+	"rag-lab.prompt-lab.page-description": "Test LLMs and prompts and see how they work together.",
+	"rag-lab.prompt-lab.page-title": "Prompt lab",
+	"rag-lab.rag-tab.configuration.search-config-label": "Select the search configurations you would like to test",
+	"rag-lab.rag-tab.configuration.search-page-navigation": "You can create more search configurations in the page",
+	"rag-lab.rag-tab.description": "Compare different search and RAG options by selecting some of your saved search configurations.",
+	"rag-lab.rag-tab.generatie-model-from-kb": "Generative model from KB settings ({{model}})",
+	"rag-lab.rag-tab.no-search-config": "You don't have any saved search configuration yet. You can create some in the page",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/en.json:816]:
```suggestion
	"rag-lab.rag-tab.no-search-config": "You don’t have any saved search configuration yet. You can create some in the page",
```

File: libs/common/src/assets/i18n/fr.json [hunk 3]
```diff
@@ -780,34 +780,41 @@
...
-	"prompt-lab.results.error": "Une erreur s'est produite lors de l'obtention de la réponse, veuillez réessayer plus tard.",
-	"prompt-lab.results.prompt": "Prompt : ",
-	"prompt-lab.results.question": "Question : {{query}}",
-	"prompt-lab.results.title": "Résultats",
+	"rag-lab.common.generate.loading.description": "Nous générons vos résultats. Merci de ne pas quitter la page.",
+	"rag-lab.common.generate.loading.title": "Juste un moment...",
+	"rag-lab.common.generate-button": "Générer",
+	"rag-lab.common.question.help": "Ajoutez votre question en appuyant sur Entrée ou sur le bouton '+'",
+	"rag-lab.common.question.label": "Question",
+	"rag-lab.common.question.title": "Écrivez jusqu'à trois questions avec lesquelles vous souhaitez tester",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:788]:
```suggestion
	"rag-lab.common.question.title": "Écrivez jusqu’à trois questions avec lesquelles vous souhaitez tester",
```

File: libs/common/src/assets/i18n/fr.json [hunk 4]
```diff
@@ -780,34 +780,41 @@
...
+	"rag-lab.common.results.download-csv": "Télécharger CSV",
+	"rag-lab.common.results.error": "Une erreur s'est produite lors de l'obtention de la réponse, veuillez réessayer plus tard.",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:790]:
```suggestion
	"rag-lab.common.results.error": "Une erreur s’est produite lors de l’obtention de la réponse, veuillez réessayer plus tard.",
```

File: libs/common/src/assets/i18n/fr.json [hunk 5]
```diff
@@ -780,34 +780,41 @@
...
+	"rag-lab.prompt-lab.configuration.models.title": "Sélectionnez tous les modèles génératifs que vous souhaitez comparer",
+	"rag-lab.prompt-lab.configuration.prompt.example": "Exemple",
+	"rag-lab.prompt-lab.configuration.prompt.examples.first-example": "En tenant compte de ce contexte :\n{context}\nDonnez une réponse détaillée à cette question :\n{question} .\nSi vous ne trouvez pas de réponse dans le contexte, dites que vous ne disposez pas de suffisamment de données.",
+	"rag-lab.prompt-lab.configuration.prompt.examples.second-example": "En tenant compte de ce contexte fourni.\n    Contexte:\n    {context}\n    Répondez à cette question comme un professionnel de la santé. :\n \"{question}\"",
+	"rag-lab.prompt-lab.configuration.prompt.examples.third-example": "En tenant compte de ce contexte fourni :\n{context} \nPouvez-vous répondre à cette question de manière claire et concise :\n {question}",
+	"rag-lab.prompt-lab.configuration.prompt.label": "Prompt",
+	"rag-lab.prompt-lab.configuration.prompt.placeholder": "Saisissez votre prompt avec les termes {question} et {context} entre parenthèses à l'endroit où vous souhaitez que la question et le contexte soient placés.",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:802]:
```suggestion
	"rag-lab.prompt-lab.configuration.prompt.placeholder": "Saisissez votre prompt avec les termes {question} et {context} entre accolades à l’endroit où vous souhaitez que la question et le contexte soient placés.",
```

File: libs/common/src/assets/i18n/fr.json [hunk 6]
```diff
@@ -780,34 +780,41 @@
...
+	"rag-lab.prompt-lab.configuration.prompt.title": "Définir les prompts qui influenceront le comportement du système lorsque vous posez une question (facultatif)",
+	"rag-lab.prompt-lab.configuration.system-prompt.examples.first-example": "Vous êtes assistant médical, utilisez la terminologie médicale",
+	"rag-lab.prompt-lab.configuration.system-prompt.examples.second-example": "Vous êtes un assistant du service client très sympathique, soyez poli",
+	"rag-lab.prompt-lab.configuration.system-prompt.examples.third-example": "Vous êtes un expert financier, utilisez des termes corrects",
+	"rag-lab.prompt-lab.configuration.system-prompt.label": "System prompt",
+	"rag-lab.prompt-lab.configuration.system-prompt.placeholder": "Définir les traits de personnalité de l'assistant.",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:808]:
```suggestion
	"rag-lab.prompt-lab.configuration.system-prompt.placeholder": "Définir les traits de personnalité de l’assistant.",
```

File: libs/common/src/assets/i18n/fr.json [hunk 7]
```diff
@@ -780,34 +780,41 @@
...
+	"rag-lab.prompt-lab.configuration.title": "Configuration",
+	"rag-lab.prompt-lab.page-description": "Testez les LLM et les prompts et voyez comment ils fonctionnent ensemble.",
+	"rag-lab.prompt-lab.page-title": "Prompt lab",
+	"rag-lab.rag-tab.configuration.search-config-label": "Sélectionnez les configurations de recherche que vous souhaitez tester",
+	"rag-lab.rag-tab.configuration.search-page-navigation": "Vous pouvez créer d'autres configurations de recherche dans la page",
+	"rag-lab.rag-tab.description": "Comparez différentes options de recherche et de RAG en sélectionnant certaines des configurations de recherche que vous avez enregistrées.",
+	"rag-lab.rag-tab.generatie-model-from-kb": "Modèle génératif issu des paramètres de la KB ({{model}})",
+	"rag-lab.rag-tab.no-search-config": "Vous n'avez pas encore de configuration de recherche sauvegardée. Vous pouvez en créer dans la page",
+	"rag-lab.rag-tab.warning.ask-to-resource-ignored": "L'option \"Demander à une ressource spécifique\" est activée dans cette configuration mais sera ignorée lors de la génération des réponses.",
```
Comment (ebrehault) [inline libs/common/src/assets/i18n/fr.json:817]:
```suggestion
	"rag-lab.rag-tab.warning.ask-to-resource-ignored": "L’option \"Demander à une ressource spécifique\" est activée dans cette configuration mais sera ignorée lors de la génération des réponses.",
```

**Review by ebrehault — COMMENTED** [review]
The RAG lab feature is not authorized for all users, the same rules used in the navbar to enable/disable the link must be applied in the search page on the RAG lab button.

File: libs/common/src/lib/rag-lab/rag-lab.component.html [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    <pa-button
+      kind="primary"
+      type="submit"
+      [disabled]="queries.length < 1 || selectedConfigs.length < 1">
+      {{ 'rag-lab.common.generate-button' | translate }}
+    </pa-button>
+  </form>
+</stf-lab-layout>
+
+<!--<div class="lab-container">-->
```
Comment (operramon) [inline libs/common/src/lib/rag-lab/rag-lab.component.html:60]: should it be removed ?

File: libs/common/src/lib/rag-lab/lab-layout/lab-layout.component.ts [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    }[]
+  > = this._results.pipe(
+    tap((result) => {
+      this.cdr.detectChanges();
+      this.updateResultsExpanderSize.next(result);
+      this.updateResultHeight();
+    }),
+  );
+  hasResults = this._results.pipe(map((results) => results.length > 0));
+  queryCollapsed: { [query: string]: boolean } = {};
```
[the remark this answers is not in the document]
Comment (mpellerin42) [reply libs/common/src/lib/rag-lab/lab-layout/lab-layout.component.ts:85]: it was used in the template to set a class, but the mechanic was updated anymore, so I cleaned it up.

**Review by ebrehault — APPROVED** [review]

**Pull request merged**